src/dosext/mfs/mfs.c

```diff
@@ -1335,16 +1336,31 @@ static struct dir_list *get_dir(char *name, char *mname, char *mext, int drive)
   char fname[8];
   char fext[3];
 
-  if(is_dos_device(name) || !find_file(name, &sbuf, drive, NULL))
+  if (is_dos_device(name)) {
+    Debug0((dbg_fd, "get_dir(): is_dos_device() returned true for '%s'\n", name));
+    if (patherror)
+      *patherror = 1;
     return NULL;
+  }
+
+  if (!find_file(name, &sbuf, drive, NULL)) {
```

Comment:
Ah, I didn't mean to split them, the
idea was to move it before the get_dir()
call, just as you did in the very first patch.